src/index.ts

```diff
@@ -203,158 +232,217 @@ function getPropertyNames(schema: Schema) {
   return schema.properties.map((item: Property) => item.name).join(', ');
 }
 
-function isEnum(prop: PropertyLike) {
-  return prop['$ref'] && prop['$ref']['enum'];
-}
-
-function isString(param: Parameter) {
-  return !param['$ref'] && param.type === 'string' && !param.format;
-}
-
 function getHandleType(prop: Parameter) {
   if (prop.contentType === 'application/json') {
     return 'char';
-  }
-  else if (isEnum(prop) && prop.contentType === 'text/plain; charset=utf-8') {
+  } else if (helpers.isEnum(prop) && prop.contentType === 'text/plain; charset=utf-8') {
     return 'char';
-  }
-  else if (!prop['$ref']) {
-    if (prop.contentType === "application/x-binary") {
-      if (prop.type === 'buffer') {
-        return 'uint8_t';
-      } else if (prop.type === 'array' && !(prop.items as Property)['$ref']) {
-        // support fixed width numeric types
-        const aprop = prop.items as Property;
-        if (aprop.type === 'number' || aprop.type === 'integer') {
-          return toCppType(aprop, '');
-        }
-      }
-    } else if (prop.contentType === 'text/plain; charset=utf-8') {
-      if (prop.type === 'string' && !prop.format) {
-        return 'char';
-      } else if (prop.type === 'buffer') {
-        return 'char';
+  } else if (prop.contentType === "application/x-binary") {
```

Comment:
(A sidenote for `xtp-bindgen`: this makes me think we might want to normalize the content type at a higher level, in case we start supporting `application/octet-stream`.)

Reply:
we maybe need a `helpers.isBinaryEncoded` like we have with json and utf8 strings. i think i just forgot to add it. that would allow us to check for any binary type.

Reply:
although semantically, if they are the same, the normalizer should collapse them into one.

Reply:
`helpers.isBinaryEncoded` sounds like a great addition and I think it would be great to normalize to `application/octet-stream` too.